modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -177,6 +178,26 @@ public final void collectMetrics(Request<?> request, Response<?> response) {
                 .map(List::size)
                 .orElse(0);
 
+            if (exceptionCount > 0) {
+                final List<Object> statusCodes = Objects.requireNonNullElse(
+                    awsRequestMetrics.getProperty(AWSRequestMetrics.Field.StatusCode),
+                    List.of()
+                );
+                // REQUESTED_RANGE_NOT_SATISFIED errors are expected errors due to RCO
+                // TODO Add more expected client error codes?
+                final long amountOfRequestRangeNotSatisfiedErrors = statusCodes.stream()
+                    .filter(e -> (Integer) e == REQUESTED_RANGE_NOT_SATISFIED.getStatus())
+                    .count();
+                if (amountOfRequestRangeNotSatisfiedErrors > 0) {
+                    s3RepositoriesMetrics.common()
+                        .requestRangeNotSatisfiedExceptionCounter()
+                        .incrementBy(amountOfRequestRangeNotSatisfiedErrors, attributes);
+                    s3RepositoriesMetrics.common()
+                        .requestRangeNotSatisfiedExceptionHistogram()
+                        .record(amountOfRequestRangeNotSatisfiedErrors, attributes);
```

Comment:
I think histogram for `RangeNotSatisfiedException` is not actually useful. The existing exception histogram was added to measure how many retries s3 client normally need to eventually succeeds. Previously we suspected that the default 3 retries is not enough and increased it to 9. The histogram helped us to be sure it was not the case since we hardly have more than 1 retry and we changed it back to 3 (https://github.com/elastic/elasticsearch-serverless/pull/1352).

`RangeNotSatisfiedException` is not retryable. I'd expect it to always be 1. The request cannot be eventually successful after seeing this error. So it's not very useful and I think we can drop it.

Reply:
That's a great point!

Reply:
I think indeed the histogram is not very useful, since I couldn't find a use case where could have multiple 416 errors for one S3 action.